packages/cmd/__tests__/CommandProcessor.test.ts

```diff
@@ -1744,6 +1746,75 @@ describe("Command Processor", () => {
         }
     });
 
+    it("should invoke the handler and process daemon response and use the environment file", async () => {
+        // Allocate the command processor
+        const processor: CommandProcessor = new CommandProcessor({
+            envVariablePrefix: ENV_VAR_PREFIX,
+            fullDefinition: SAMPLE_COMPLEX_COMMAND,
+            definition: SAMPLE_COMMAND_REAL_HANDLER,
+            helpGenerator: FAKE_HELP_GENERATOR,
+            profileManagerFactory: FAKE_PROFILE_MANAGER_FACTORY,
+            rootCommandName: SAMPLE_ROOT_COMMAND,
+            commandLine: "",
+            promptPhrase: "dummydummy",
+            daemonContext: {
+                response: {
+                    cwd: process.cwd(),
+                    env: { UNIT_TEST_ENV: "new" }
+                }
+            }
+        });
+
+        // Mock read stdin
+        (SharedOptions.readStdinIfRequested as any) = jest.fn((args, response, type) => {
+            // Nothing to do
+        });
+
+        // Mock the profile loader
+        (CommandProfileLoader.loader as any) = jest.fn((args) => {
+            return {
+                loadProfiles: (profArgs: any) => {
+                    // Nothing to do
+                }
+            };
+        });
+
+        jest.spyOn(process, "chdir");
+        const mockConfigReload = jest.fn();
+        jest.spyOn(ImperativeConfig, "instance", "get").mockReturnValue({
+            config: { reload: mockConfigReload }
+        } as any);
+        jest.spyOn(EnvFileUtils, "getEnvironmentFilePath").mockReturnValueOnce(join(__dirname, "__resources__", ".zowe.env.json"));
+
+        const parms: any = {
+            arguments: {
+                _: ["check", "for", "banana"],
+                $0: "",
+                valid: true
+            },
+            silent: true
+        };
+        EnvFileUtils.setEnvironmentForApp("zowe", false);
+        process.env.UNIT_TEST_ENV = "old";
+        try {
+            const envVarCount = Object.keys(process.env).length;
+            const commandResponse: ICommandResponse = await processor.invoke(parms);
+            expect(commandResponse).toBeDefined();
+            expect(commandResponse).toMatchSnapshot();
```

Comment:
I'm curious what the purpose of using snapshots here is. It looks like the command response is just an empty object?

Not requesting changes as there are other tests in this file that take snapshots for the same exact empty command response object

Reply:
I just copied it from the above test, I am not sure either. I can remove it if you prefer.

Reply:
no need 😋